modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/internal/exportimport/data/handler/KBArticleStagedModelDataHandler.java

```diff
@@ -463,12 +465,47 @@ private void _importKBArticleAttachments(
 		serviceContext.setCompanyId(portletDataContext.getCompanyId());
 		serviceContext.setScopeGroupId(portletDataContext.getScopeGroupId());
 
+		Map<String, FileEntry> attachments = new HashMap<>();
+
+		for (FileEntry attachment :
+				importedKBArticle.getAttachmentsFileEntries()) {
+
+			attachments.put(attachment.getUuid(), attachment);
+		}
+
 		for (Element dlFileEntryElement : dlFileEntryElements) {
 			String path = dlFileEntryElement.attributeValue("path");
 
 			FileEntry fileEntry =
 				(FileEntry)portletDataContext.getZipEntryAsObject(path);
 
+			if (attachments.get(fileEntry.getUuid()) != null) {
+				attachments.remove(fileEntry.getUuid());
+
+				continue;
+			}
+
+			DLFileEntry importedFileEntry =
+				_dlFileEntryLocalService.getFileEntryByUuidAndGroupId(
+					fileEntry.getUuid(), portletDataContext.getScopeGroupId());
+
+			if (importedFileEntry != null) {
```

Comment:
You are using "getFileEntryByUuidAndGroupId" so if the file doesn't exist a NoSuchFile will the thown. 

I think that this null check is not necessary (or you have to use a fetch method)